source/ai-server-logger.js

```diff
@@ -0,0 +1,122 @@
+(function() {
+    'use strict';
+
+    angular.module('ai.public.serverLogger', ['ng'])
+        .config(function($provide) {
+            $provide.decorator('$log', function($delegate, $injector) {
+                var logger = $injector.get('AiServerLogger');
+                var delegateError = $delegate.error;
+                var delegateWarn = $delegate.warn;
+                var delegateInfo = $delegate.info;
+                var delegateDebug = $delegate.debug;
+
+                var enhanceLoggingFunction = function(loggingFunction, severity) {
+                    return function() {
+                        var args = [].slice.call(arguments);
+                        loggingFunction.apply(null, args);
+
+                        var message = args.length === 1 ? _.first(args) : JSON.stringify(args);
+                        logger.send(message, severity);
+                    };
+                };
+
+                $delegate.error = enhanceLoggingFunction(delegateError, 'Error');
+                $delegate.warn = enhanceLoggingFunction(delegateWarn, 'Warning');
+                $delegate.info = enhanceLoggingFunction(delegateInfo, 'Info');
+                $delegate.debug = enhanceLoggingFunction(delegateDebug, 'Debug');
+                $delegate.trace = enhanceLoggingFunction(delegateDebug, 'Trace');
+
+                return $delegate;
+            });
+        })
+        .provider('AiServerLogger', function() {
+            var headers = {};
+            var serverPostEndPoint;
+            var queueSize = 100;
+            var postToServerDelay = 1000 * 60 * 10;  //10 minutes
+
+            var transformParameters = function() {
+                return {
+                    Message: arguments[0],
+                    Level: arguments[1]
+                };
+            };
+
+            var ServerLogger = (function() {
+                function ServerLogger(loggerConfig) {
+                    var config = loggerConfig;
+                    var queue = [];
+                    var timeOutId;
```

Comment:
timeout seems to be one word :) so no camel casing

https://en.wikipedia.org/wiki/Timeout_(computing)